### PR TITLE
Update test suite to use pytest, remove related in setup.py

### PR DESCRIPTION
**Pull request by kwongtn**
Setuptools removes use of `setuptools.command.test`
This PR serves as a quick temporary fix.

Refer:
- https://setuptools.pypa.io/en/stable/history.html#v72-0-0

### CODE REVIEW

**Comment by wongjiahau** [comment]
Running `poetry add git+https://github.com/kwongtn/django-celery-beat#restrict-setuptools-to-max-v71` still fails with the same error

**Comment by cclauss** [comment]
Nice initiative!!  This is still happening on our GitHub Actions Py3.12 tests...
> ModuleNotFoundError: No module named 'setuptools.command.test'

**Comment by kwongtn** [comment]
If anyone wants to take over and continue fixing, please do so..
I'm out of ideas 😥

**Comment by paretosti** [comment]
@wongjiahau 
> Running `poetry add git+https://github.com/kwongtn/django-celery-beat#restrict-setuptools-to-max-v71` still fails with the same error

Is this the right way to install a branch? Is it not
```
poetry add git+https://github.com/kwongtn/django-celery-beat/restrict-setuptools-to-max-v71
```

File: setup.py [hunk 1]
```diff
@@ -6,7 +6,6 @@
 import sys
```
Comment (cclauss) [inline setup.py:6]:
```suggestion
import pytest
```

**Comment by shijl0925** [comment]
pip3 install django-celery-beat@git+https://github.com/kwongtn/django-celery-beat.git@fix/restrict-setuptools-to-max-v71

Collecting django-celery-beat@ git+https://github.com/kwongtn/django-celery-beat.git@fix/restrict-setuptools-to-max-v71
  Cloning https://github.com/kwongtn/django-celery-beat.git (to revision fix/restrict-setuptools-to-max-v71) to /tmp/pip-install-otwx_qhh/django-celery-beat_a2925745ef8d482b967cd8ec9c642a08
  Running command git clone --filter=blob:none --quiet https://github.com/kwongtn/django-celery-beat.git /tmp/pip-install-otwx_qhh/django-celery-beat_a2925745ef8d482b967cd8ec9c642a08
  Running command git checkout -b fix/restrict-setuptools-to-max-v71 --track origin/fix/restrict-setuptools-to-max-v71
  Switched to a new branch 'fix/restrict-setuptools-to-max-v71'
  branch 'fix/restrict-setuptools-to-max-v71' set up to track 'origin/fix/restrict-setuptools-to-max-v71'.
  Resolved https://github.com/kwongtn/django-celery-beat.git to commit 5a21c38679f416dff525754a6f27d7909877e7c8
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-i22alft_/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 327, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-i22alft_/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 297, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-i22alft_/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 497, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-i22alft_/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 313, in run_setup
          exec(code, locals())
        File "<string>", line 123, in <module>
      NameError: name 'pytest' is not defined. Did you mean: 'bytes'?
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.

the code should be removed?
```
cmdclass={'test': pytest}
```

**Comment by kwongtn** [comment]
@shijl0925 Yup I think you're right -- removed it and it works nicely now

Not sure how it will affect the test in this repo tho 😅
But I replaced it for pytest to be called directly from terminal instead of being invoked by setup.py 

I'll be locking my prod build to 8d30439 , but @cclauss  feel free to make required edits

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -43,7 +43,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install tox tox-gh-actions
+        python -m pip install tox tox-gh-actions pytest
```
Comment (cclauss) [inline .github/workflows/test.yml:46]: Is this needed outside of `tox.ini` instead of in the `deps` inside that file?

**Review by cclauss — CHANGES_REQUESTED** [review]

File: requirements/pkgutils.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-setuptools>=40.8.0
+setuptools>=40.8.0,<72
```
[the remark this answers is not in the document]
Comment (kwongtn98) [reply requirements/pkgutils.txt]: Not anymore, removed in 65d5743818a447a7f775128bfac50c729bb54c75

File: setup.py [hunk 4]
```diff
@@ -134,7 +119,6 @@ def run_tests(self):
     python_requires='>=3.8',
     install_requires=reqs('default.txt') + reqs('runtime.txt'),
     tests_require=reqs('test.txt') + reqs('test-django.txt'),
```
[the remark this answers is not in the document]
Comment (kwongtn98) [reply setup.py]: You're right 😁
Removed in 03d421cc4417b9c120c9c03c06c9404a9c9c7e2f

File: .github/workflows/test.yml (hunk 2 above)
Comment (kwongtn98) [reply .github/workflows/test.yml]: Moved 😁
1f013413bbbeaa03f9c377297c865ac633ceae6a

**Comment by cclauss** [comment]
Please try `make test` on your computer.

File: Makefile [hunk 5]
```diff
@@ -139,7 +139,7 @@ test-all: clean-pyc
 	$(TOX)
 
 test:
-	$(PYTHON) setup.py test
+	$(PYTHON) -m $(PYTEST)
```
[the remark this answers is not in the document]
Comment (kwongtn) [reply Makefile:142]: Unfortunately no, but `tox` does though
Although some tests fail 😅

**Review by cclauss — APPROVED** [review]
@kwongtn. Awesome work!  Thanks for doing this.

@Nusnus @auvipy We will need a new release to PyPI to cover the breaking change in `setuptools`.
Also see:
* celery/celery#9157

**Pull request merged**